common/src/main/java/com/wynntils/mc/mixin/ClientPackSourceMixin.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.mixin;
+
+import com.google.common.hash.Hashing;
+import com.google.common.io.Files;
+import com.wynntils.mc.EventFactory;
+import java.io.IOException;
+import net.minecraft.client.resources.ClientPackSource;
+import net.minecraft.server.packs.AbstractPackResources;
+import net.minecraft.server.packs.PackResources;
+import net.minecraft.server.packs.repository.Pack;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(ClientPackSource.class)
+public abstract class ClientPackSourceMixin {
+    @Shadow
+    private Pack serverPack;
+
+    @Inject(method = "clearServerPack", at = @At("HEAD"), cancellable = true)
+    private void onClearServerPackPre(CallbackInfo ci) {
+        if (serverPack == null) {
```

Comment:
There's just a tad too much going on here. Don't you think the received of the event can calculate the hash, for instance?

Reply:
(The point in doing it after the event is sent is that it will not affect non-Wynncraft servers.)

Reply:
Sure

Reply:
We always need to send the event because this happens after disconnect. Do you still want to move this?

Reply:
Ah. Hm. Ok, keep it. Maybe add a comment stating what you just said, that this happens after disconnect so needs to be done always.